### PR TITLE
Produce only "und" data for metazone periods

### LINKED ISSUE #2257: MetaZonePeriod creates duplicate Json files for different locale
MetaZonePeriod creates duplicate Json files for different locale, we only need one of them for optimization.

### PR DESCRIPTION
**Pull request by sffc**
Fixes #2257

Also refactors this section of the transformer code a bit.

### CODE REVIEW

File: provider/datagen/src/transform/cldr/time_zones/mod.rs [hunk 1]
```diff
@@ -92,13 +93,18 @@ macro_rules! impl_data_provider {
...
             impl IterableDataProvider<$marker> for TimeZonesProvider {
                 fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
-                    Ok(self
-                        .source
-                        .cldr()?
-                        .dates("gregorian")
-                        .list_langs()?
-                        .map(DataLocale::from)
-                        .collect())
+                    if <$marker>::KEY == MetaZonePeriodV1Marker::KEY {
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/time_zones/mod.rs:96]: Note: Actually this code here is the only part strictly necessary. The rest is cleanup.

Comment (samchen61661) [reply provider/datagen/src/transform/cldr/time_zones/mod.rs:96]: Great! Good to know that!

**Review by samchen61661 — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]
question: should this data struct live in the icu_timezones crate? Are there any others that should as well? I assume anything that is `und`-based probably should live there eventually.

**Comment by sffc** [comment]
> question: should this data struct live in the icu_timezones crate? Are there any others that should as well? I assume anything that is `und`-based probably should live there eventually.

Yes; good suggestion. The other time zone keys are all display names, but this one is for calculations only.

**Pull request merged**